protos/atomic_swap.proto

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 REMME
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+syntax = "proto3";
+
+message AtomicSwapMethod {
+    enum Method {
+        INIT = 0;
+        APPROVE = 1;
+        EXPIRE = 2;
+        SET_SECRET_LOCK = 3;
+        CLOSE = 4;
+    }
+}
+
+message AtomicSwapInitPayload {
+    string receiver_address = 1;
+    string sender_address_non_local = 7;
+    uint64 amount = 2;
+    string swap_id = 3;
+    string secret_lock_optional_bob = 4;
+    string email_address_encrypted_optional_alice = 5;
+    uint32 created_at = 6;
+}
+
+message AtomicSwapApprovePayload {
+    string swap_id = 1;
+}
+
+message AtomicSwapExpirePayload {
+    string swap_id = 1;
+}
+
+message AtomicSwapSetSecretLockPayload {
+    string swap_id = 1;
+    string secret_lock = 2;
+}
+
+message AtomicSwapClosePayload {
+    string swap_id = 1;
+    string secret_key = 2;
+}
+
+message AtomicSwapInfo {
+    bool is_closed = 1;
+    bool is_approved = 11;
+    string sender_address = 2;
+    string sender_address_non_local = 12;
+    string receiver_address = 3;
+    uint64 amount = 4;
+    string email_address_encrypted_optional = 5;
```

Comment:
It is made to pass encrypted email address to bot for notification purposes

Reply:
I'm not sure if we need this on blockchain.

Reply:
Still don't see why should we store this data on the blockchain? This is just a piece of additional work for the blockchain and this can be stored in any conventional DB.